### PR TITLE
Merge with upstream 4.0.3

### PR DESCRIPTION
**Pull request by aruniverse**
only conflicts/changes required were with the following files:

- packages\react-scripts\package.json
- packages\react-scripts\config\webpack.config.js
- packages\react-scripts\scripts\build.js
- packages\react-scripts\scripts\utils\createJestConfig.js
- packages\react-scripts\README-imodeljs.md

Diff between upstream and our imjs fork: https://github.com/imodeljs/create-react-app/compare/master...imodeljs:sync/4.0.3
Diff between our cra fork 3.x and 4.0 (this pr): https://github.com/imodeljs/create-react-app/compare/3.x...imodeljs:sync/4.0.3

Note, the only real change betwween 3.x and 4.0.3 is https://github.com/imodeljs/create-react-app/commit/41a57a9d4d91219591c865d4be0dcc38d1a7a071, which bumps the ver of ts-jest

### CODE REVIEW

**Comment by calebmshafer** [comment]
@aruniverse this looks great. Thanks for picking it up!

Definitely agree that we need a release branch for 3.x as we will/should continue to support it for iModel.js as it doesn't look like there is anything incompatible. Is that your impression as well?

However, the only reason I see people not doing it is the requirement to move up to ESLint 7.x, whereas iModel.js supports both 6.x and 7.x.  Any others?

I do want to see why all of the tests fail again :) I really dislike all the red but not needed for this PR.

I'll give it a test run this weekend with the iTwin Viewer CRA template to see how things go.

**Comment by calebmshafer** [comment]
@aruniverse, quick update. I merged the other PR for 3.x and then started testing this one out. Other than the iModel.js repo, I tried it with the Desktop Starter.  Everything appears to be working well.

I haven't created the release branch yet for 3.x but I was going to name it `imodeljs/3.x`.

**Comment by aruniverse** [comment]
> Definitely agree that we need a release branch for 3.x as we will/should continue to support it for iModel.js as it doesn't look like there is anything incompatible. Is that your impression as well?

Yea that's my impression as well, imo the major ver bump to cra 4.0 is relatively minor compared to bumping imjs versions, where most of the changes on upstreams end _should_ be compatible with what users have now. Might be an issue if users are pinning down their `@types/react` version to an earlier version and are using `react@16.14.0` because of missing types for the new jsx-transform. (this is what happened in core, but able to get around this by disabling the transform)

> However, the only reason I see people not doing it is the requirement to move up to ESLint 7.x, whereas iModel.js supports both 6.x and 7.x. Any others?

If consumers are using jest, they'll need to bump to atleast ts v3.8, but hopefully thats pretty minor too.

Highlight of changes in upstream 4.0: https://github.com/facebook/create-react-app/releases/tag/v4.0.0
https://reactjs.org/blog/2020/10/20/react-v17.html
https://www.typescriptlang.org/docs/handbook/release-notes/typescript-4-0.html
https://eslint.org/docs/user-guide/migrating-to-7.0.0
https://jestjs.io/blog/2020/05/05/jest-26
https://jestjs.io/blog/2020/01/21/jest-25

**Comment by calebmshafer** [comment]
@aruniverse looks like there hasn't been any other releases from CRA since this one.  I'm going to go ahead and create the `imodeljs/3.x` branch now from the latest 3.4.11 release.  Then we can go from there.

**Comment by calebmshafer** [comment]
@aruniverse lets go ahead and merge this PR this week. I have created the 3.x branch for all future releasing so this should be clear to merge.

We still need to figure out the issue with the latest change to the copy webpack plugin. That is still holding up the official usage of it in the imodeljs repo.  Have yo made any progress on it? If not, maybe @wgoehrig and I can help this week.

**Comment by aruniverse** [comment]
> We still need to figure out the issue with the latest change to the copy webpack plugin. That is still holding up the official usage of it in the imodeljs repo. 

Yea i was just thinking about this, and pinged you on the side haha
I haven't made any progress on it since i returned for vacation. Will return to investigating it tomorrow, and most likely creating an issue in that pkg's repo

**Comment by calebmshafer** [comment]
@aruniverse sounds good, I'm thinking we just back out the update for now or make it optional behind a flag.

**Review by calebmshafer — APPROVED** [review]
I think the process we laid out before is to make a merge commit here rather than squash so let me know when you're ready.

I'd say just remove the -dev and we're set

File: packages/react-scripts/package.json
Comment (aruniverse) [inline packages/react-scripts/package.json:3]:
```suggestion
  "version": "4.0.3",
```

**Pull request merged**
